plugins/scaffolder-backend/src/scaffolder/stages/publish/publishers.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import { parseLocationAnnotation } from '../helpers';
+import { PublisherBase, PublisherBuilder } from './types';
+import { RemoteProtocol } from '../types';
+
+export class Publishers implements PublisherBuilder {
+  private publisherMap = new Map<RemoteProtocol, PublisherBase>();
+
+  register(protocol: RemoteProtocol, publisher: PublisherBase) {
+    this.publisherMap.set(protocol, publisher);
+  }
+
+  get(template: TemplateEntityV1alpha1): PublisherBase {
+    const { protocol } = parseLocationAnnotation(template);
+    const publisher = this.publisherMap.get(protocol);
```

Comment:
I have a small concern here that when using the `file:` location in the catalog, this will break as there is no `file:` publisher registered. Maybe we can fallback to the `github` one for now. And throw an error if that doesn't exist.

It's a little hard without doing what's mentioned here:  https://github.com/spotify/backstage/pull/2426/files#r489404507

Reply:
It would also be good to document this new behaviour in https://github.com/spotify/backstage/blob/master/docs/features/software-templates/installation.md and documention the above behaviour

Reply:
I believe that's already covered by us registering the GitHub publisher for the `file` protocol: https://github.com/spotify/backstage/pull/2426/files#diff-b7b27027cea69fa63c131d6e290c9156R59